utils_nlp/models/transformers/common.py

```diff
@@ -135,9 +150,9 @@ def fine_tune(
             ]
             optimizer = AdamW(optimizer_grouped_parameters, lr=learning_rate, eps=adam_epsilon)
 
-        if scheduler is None:
-            scheduler = get_linear_schedule_with_warmup(optimizer, num_warmup_steps=warmup_steps, num_training_steps=t_total)
 
+            if t_total != -1 and scheduler is None:
```

Comment:
The change for the common is to make sure it's compatible with the current notebooks. Otherwise, this pr includes too many changes at one time.

Reply:
OK. Got it. My thought is the user may use an optimizer that's compatible with get_linear_schedule_with_warmup. Maybe we can print out a warning here as well, saying no scheduler is applied to custom optimizer. We can add an option for a custom scheduler if there is a need in the future. 
Can you address my second comment as well? A warning message would be sufficient.

Reply:
for your second comment, I don't get this part "If the scheduler is None because max_steps is not set and the data size is unknown".  The scheduler is by default is None. If users provide a scheduler, that scheduler is used.  If Scheduler is not set, and when t_total != -1 , a scheduler is created.  The only time that a scheduler will not be created is when max_step is not set and ( we can't get dataset length  or num_train_epochs is -1). I agree that we should provide a warning message in aforementioned case.

Reply:
Yes. That's what I meant. The user may assume setting warmup_steps and weight_decay will guarantee learning rate warmup and decay to happen. So it's better to warn them.
On a second thought, if max_steps is not set, dataset length is unknown, num_train_epochs is -1, will the code go into an infinite loop?